Peoplesoft/modules/compute/compute.tf

```diff
@@ -0,0 +1,171 @@
+  count                         = "${var.create_av_set ? 0 : 1}"
+  location                      = "${var.location}"
+  resource_group_name           = "${var.resource_group_name}"
+  vm_size                       = "${var.vm_size}"
+  network_interface_ids         = ["${element(azurerm_network_interface.compute.*.id, count.index)}"]
+  delete_os_disk_on_termination = "true"
+
+  # Add cloud init
+  storage_image_reference {
+    publisher = "${var.os_publisher}"
+    offer     = "${var.os_offer}"
+    sku       = "${var.os_sku}"
+    version   = "${var.os_version}"
+}
+
+  storage_os_disk {
+    name              = "${var.compute_hostname_prefix}-${format("%.02d",count.index + 1)}-disk-OS"
+    create_option     = "FromImage"
+    caching           = "ReadWrite"
+    disk_size_gb      = "${var.compute_boot_volume_size_in_gb}"
+    managed_disk_type = "${var.storage_account_type}"
+  }
+
+  os_profile {
+    computer_name  = "${var.compute_hostname_prefix}-${format("%.02d",count.index + 1)}"
+    admin_username = "${var.admin_username}"
+    admin_password = "${var.admin_password}"
+    custom_data    = "${var.custom_data}"
+  }
+
+  os_profile_linux_config {
+    disable_password_authentication = true
+
+    ssh_keys {
+      path     = "/home/${var.admin_username}/.ssh/authorized_keys"
+      key_data = "${file("${var.compute_ssh_public_key}")}"
+    }
+  }
+
+  tags = "${var.tags}"
+
+  boot_diagnostics {
+    enabled     = "true"
+    storage_uri = "${var.boot_diag_SA_endpoint}"
+  }
+}
+resource "azurerm_managed_disk" "vm_data_disks" {
+    name                 = "${var.compute_hostname_prefix}-${format("%.02d",count.index + 1)}-disk-data-01"  
+    location             = "${var.location}"
+    resource_group_name  = "${var.resource_group_name}"
+    storage_account_type = "${var.storage_account_type}"
+    create_option        = "Empty"
+    disk_size_gb         = "${var.data_disk_size_gb}"
+    count                = "${var.create_data_disk}"
+
+}
+
+resource "azurerm_virtual_machine_data_disk_attachment" "vm_data_disks_attachment" {
+  managed_disk_id    = "${element(azurerm_managed_disk.vm_data_disks.*.id, count.index)}"
+  virtual_machine_id = "${element(azurerm_virtual_machine.compute.*.id, count.index)}"
+  lun                = "${count.index}"
+  caching            = "None"
+  count = "${var.create_data_disk}"
+
+
+ provisioner "remote-exec" {
+ inline = [
+ "sudo su", # TODO: Upload private key to the machine too. Figure out how. 
+ "yum install -y oracle-ebs-server-R12-preinstall.x86_64", # This will be replaced by cloud-init for Oracle Linux 7.7+ 
+ # "echo 'type=83' | sfdisk /dev/sdc", #sfdisk alternative to fdisk
+ "echo n\np\n1\n\n\np\nw | fdisk /dev/sdc", # n=> New parition, p=> primary partition, 1 => default partition number 1, "" => default first and last sector, w => write to partition table #TODO: Check if \n works
+ "mkfs -t ext4 /dec/sdc1",
+ "chown -R ${var.admin_username} /u01"
```

Comment:
This didn't work for me. I'm trying to solve this using azurerm_virtual_machine_extension now. I'll keep you posted on this. No action for you at the moment.